src/sentry/tasks/post_process.py

```diff
@@ -116,38 +116,57 @@ def handle_owner_assignment(project, group, event):
 
 
 @instrumented_task(name="sentry.tasks.post_process.post_process_group")
-def post_process_group(event, is_new, is_regression, is_new_group_environment, **kwargs):
+def post_process_group(
+    event, is_new, is_regression, is_new_group_environment, cache_key=None, group_id=None, **kwargs
+):
     """
     Fires post processing hooks for a group.
     """
-    set_current_project(event.project_id)
-
+    from sentry.eventstore.models import Event
+    from sentry.eventstore.processing import event_processing_store
     from sentry.utils import snuba
     from sentry.reprocessing2 import is_reprocessed_event
 
     with snuba.options_override({"consistent": True}):
-        if is_reprocessed_event(event.data):
+        # The event parameter will be removed after transitioning to
+        # event_processing_store is complete.
+        if cache_key and event is None:
+            data = event_processing_store.get(cache_key)
+            if not data:
+                logger.info(
+                    "post_process.skipped",
+                    extra={"cache_key": cache_key, "reason": "missing_cache"},
+                )
+                return
+            event = Event(
+                project_id=data["project"], event_id=data["event_id"], group_id=group_id, data=data
+            )
+        elif check_event_already_post_processed(event):
+            if cache_key:
+                event_processing_store.delete_by_key(cache_key)
```

Comment:
This was missing a check for cache_key which caused the problems last time around. I had previously thought that replays were rare and only happened when consumers crashed or we manually replayed. That is not true and replays happen very frequently.

Reply:
Is this the only change from the previous pr? If so, lgtm.

Reply:
Yeah this is the important difference from before.